src/mpc-hc/MainFrm.cpp

```diff
@@ -12703,19 +12818,31 @@ void CMainFrame::OpenSetupWindowTitle(bool reset /*= false*/)
                 }
                 if (!use_label) {
                     title = GetFileName();
+                    bool hasName = false;
 
                     if (s.fTitleBarTextTitle) {
                         BeginEnumFilters(m_pGB, pEF, pBF) {
                             if (CComQIPtr<IAMMediaContent, &IID_IAMMediaContent> pAMMC = pBF) {
                                 CComBSTR bstr;
                                 if (SUCCEEDED(pAMMC->get_Title(&bstr)) && bstr.Length()) {
                                     title = CString(bstr.m_str);
+                                    hasName = true;
                                     break;
                                 }
                             }
                         }
                         EndEnumFilters;
                     }
+
+                    if (!hasName && pli->m_cue) {

```

Comment:
I think it is better to remove the m_cue_metadata global variable and also this piece of code. This same title information is also set as the label of the playlist entry, right?

We should follow the option to use title (Options > Player > Title bar > replace filename with title). I might enable that by default in another patch.

So higher up this change could be made:
```
bool use_label = false;
CPlaylistItem* pli = m_wndPlaylistBar.GetCur();
if (pli && pli->m_label && !pli->m_label.IsEmpty()) {
    if (s.fTitleBarTextTitle || pli->m_fns.GetHead().Left(4) == _T("http")) {
        title = pli->m_label;
        use_label = true;
    }
}
```

Reply:
Seems a good idea.